src/devdocs2zim/generator.py

```diff
@@ -2,17 +2,29 @@
 import datetime
 import os
 import re
+import tempfile
 from collections import defaultdict
 from pathlib import Path
 
 from bs4 import BeautifulSoup
 from jinja2 import Environment, FileSystemLoader, select_autoescape
+from PIL import UnidentifiedImageError
 from pydantic import BaseModel
 from zimscraperlib.constants import (  # pyright: ignore[reportMissingTypeStubs]
     MAXIMUM_DESCRIPTION_METADATA_LENGTH,
     MAXIMUM_LONG_DESCRIPTION_METADATA_LENGTH,
     RECOMMENDED_MAX_TITLE_LENGTH,
 )
+from zimscraperlib.image.conversion import (  # pyright: ignore[reportMissingTypeStubs]
```

Comment:
I don't get why you need all these `pyright` annotations, recent zimscraperlib has proper type annotations now. Isn't that only an issue with your local setup? Probably worth to try to remove them and check if the CI complains, I don't need these anymore on other projects I have (but I might miss something).

Reply:
I don't think it's just a local thing, GitHub actions is complaining too:

https://github.com/openzim/devdocs/actions/runs/11716568099/job/32634832669?pr=37

Semi-related, I noticed the `.pre-commit-config.yaml` files aren't using hatch, that might be something to look at more broadly to try and drive out a little more inconsistency.

Reply:
Indeed, this is linked to the `strict` mode of pyright, in other scraper I've used I'm only in `basic`. 

I've opened https://github.com/openzim/python-scraperlib/issues/212 to track this issue, we should be able to create a scraper in `strict` mode without having issues on our own scraperlib. 

Regarding the move of pre-commit to hatch, I didn't knew it was possible. Definitely something very valuable, I agree, especially as a maintainer it is a pain to always update dependencies version both in `pyproject.toml` and in `.pre-commit-config.yaml`. I've opened https://github.com/openzim/_python-bootstrap/issues/51

Reply:
Do you want me to bump down the version here? In general I like strict checks for the larger projects I work on because it reduces some of the mental burden.

However, this is a smaller project so it might be more valuable to instead reduce the switching cost from one scraper to another.

I don't really care one way or the other :-)

Reply:
Please keep `strict`, it is our "recommendation" whenever feasible, I'm glad you used it ^^